src/Controller/ModulesController.php

```diff
@@ -302,6 +302,41 @@ public function save() : ?Response
         ]);
     }
 
+    /**
+     * Create new object from ajax request.
+     *
+     * @return void
+     */
+    public function saveJson() : void
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->prepareRequest($this->objectType);
+
+        try {
+            // upload file (if available)
+            $this->Modules->upload($requestData);
+
+            // save data
+            $response = $this->apiClient->save($this->objectType, $requestData);
+        } catch (InternalErrorException | BEditaClientException | UploadException $error) {
```

Comment:
We should probably catch only `BEditaClientException` other exceptions should not be thrown here but handled at higher level..